packages/web-components/fast-foundation/src/number-field/number-field.template.ts

```diff
@@ -0,0 +1,75 @@
+import { html, ref, slotted } from "@microsoft/fast-element";
+import { endTemplate, startTemplate } from "../patterns";
+import { NumberField } from "./number-field";
+
+/**
+ * The template for the {@link @microsoft/fast-foundation#(NumberField:class)} component.
+ * @public
+ */
+export const NumberFieldTemplate = html<NumberField>`
+    <template
+        class="
+            ${x => (x.readOnly ? "readonly" : "")}
+        "
+    >
+        <label
+            part="label"
+            for="control"
+            class="${x =>
+                x.defaultSlottedNodes && x.defaultSlottedNodes.length
+                    ? "label"
+                    : "label label__hidden"}"
+        >
+            <slot ${slotted("defaultSlottedNodes")}></slot>
+        </label>
+        <div class="root" part="root">
+            ${startTemplate}
+            <input
+                class="control"
+                part="control"
+                id="control"
+                @input="${x => x.handleTextInput()}"
+                @change="${x => x.handleChange()}"
+                ?autofocus="${x => x.autofocus}"
+                ?disabled="${x => x.disabled}"
+                list="${x => x.list}"
+                maxlength="${x => x.maxlength}"
+                minlength="${x => x.minlength}"
+                placeholder="${x => x.placeholder}"
+                ?readonly="${x => x.readOnly}"
+                ?required="${x => x.required}"
+                size="${x => x.size}"
+                :value="${x => x.value}"
+                type="text"
+                min="${x => x.min}"
+                max="${x => x.max}"
+                step="${x => x.step}"
+                aria-atomic="${x => x.ariaAtomic}"
+                aria-busy="${x => x.ariaBusy}"
+                aria-controls="${x => x.ariaControls}"
+                aria-current="${x => x.ariaCurrent}"
+                aria-describedBy="${x => x.ariaDescribedby}"
+                aria-details="${x => x.ariaDetails}"
+                aria-disabled="${x => x.ariaDisabled}"
+                aria-errormessage="${x => x.ariaErrormessage}"
+                aria-flowto="${x => x.ariaFlowto}"
+                aria-haspopup="${x => x.ariaHaspopup}"
+                aria-hidden="${x => x.ariaHidden}"
+                aria-invalid="${x => x.ariaInvalid}"
+                aria-keyshortcuts="${x => x.ariaKeyshortcuts}"
+                aria-label="${x => x.ariaLabel}"
+                aria-labelledby="${x => x.ariaLabelledby}"
+                aria-live="${x => x.ariaLive}"
+                aria-owns="${x => x.ariaOwns}"
+                aria-relevant="${x => x.ariaRelevant}"
+                aria-roledescription="${x => x.ariaRoledescription}"
+                ${ref("control")}
+            />
+            <div class="controls">
+                <div class="step_up" @click="${x => x.stepUp()}"></div>
```

Comment:
nit: I think we are trying to stick with kabob-case for multiple word classes that aren't things like `__hidden`.
This at least seems to be the pattern for *most* other multiple word classes on our components :P

Reply:
Good to know. I was following the pattern based on label__hidden. I've updated these to step-up and step-class.

Reply:
Yeah I am sure there are other opinions, which is why I say nit, no big deal.